In [1]:
import pandas as pd
import datetime

posts_df = pd.read_csv("https://raw.githubusercontent.com/anxiousasallhelll/exINCult/main/Worship_Service_Examination_Posts.csv")
posts_df

Title      ID        Date  \
0    Weekend worship service examination (Feb 01 - ...  ex0kyh  2020-01-31   
1    Weekend Worship Service Examination (Apr 24 - ...  g725k9  2020-04-23   
2    Weekend Worship Service Examination (May 29 - ...  gsm82c  2020-05-28   
3    Mid-week Worship Service Examination (Dec 07 -...  rb11qd  2020-05-28   
4    Mid-week Worship Service Examination (Nov 30 -...  r5p3h8  2020-05-28   
..                                                 ...     ...         ...   
195  Weekend Worship Service Examination (Jun 24 - ...  vjp4h9  2022-06-24   
196  Weekend Worship Service Examination (Jul 01 - ...  vp1iuf  2022-07-01   
197  Weekend Worship Service Examination (Jul 08 - ...  vubg58  2022-07-08   
198  Weekend Worship Service Examination (Jul 15 - ...  vzpftx  2022-07-15   
199  Weekend Worship Service Examination (Jul 22 - ...  w5bck2  2022-07-22   

     Total Comments                                           Post URL  
0                33  https://www.reddit.com/r/exIglesiaNiCristo/com...  
1                34  https://www.reddit.com/r/exIglesiaNiCristo/com...  
2                16  https://www.reddit.com/r/exIglesiaNiCristo/com...  
3                15  https://www.reddit.com/r/exIglesiaNiCristo/com...  
4                11  https://www.reddit.com/r/exIglesiaNiCristo/com...  
..              ...                                                ...  
195              22  https://www.reddit.com/r/exIglesiaNiCristo/com...  
196              25  https://www.reddit.com/r/exIglesiaNiCristo/com...  
197              19  https://www.reddit.com/r/exIglesiaNiCristo/com...  
198              28  https://www.reddit.com/r/exIglesiaNiCristo/com...  
199              29  https://www.reddit.com/r/exIglesiaNiCristo/com...  

[200 rows x 5 columns]

In [2]:
top_posts_df = posts_df.drop(columns=['ID'])
top_15_posts_df = top_posts_df.sort_values(by='Total Comments', ascending=False).head(15)
top_15_posts_df

Title        Date  \
181  Weekend Worship Service Examination (Feb 04 - ...  2022-02-04   
37   Mid-week Worship Service Examination (Feb 01 -...  2020-05-28   
102  Mid-week Worship Service Examination (Jun 07 -...  2020-05-28   
47   Mid-week Worship Service Examination (Feb 08 -...  2020-05-28   
92   Mid-week Worship Service Examination (Apr 28 -...  2020-05-28   
193  Weekend Worship Service Examination (Jun 10 - ...  2022-06-10   
105  Weekend Worship Service Examination (Jul 24 - ...  2020-07-23   
188  Weekend Worship Service Examination (Mar 25 - ...  2022-03-25   
182  Weekend Worship Service Examination (Feb 11 - ...  2022-02-11   
59   Mid-week worship service examination (Jan 15 -...  2020-05-28   
185  Weekend Worship Service Examination (Mar 04 - ...  2022-03-04   
183  Weekend Worship Service Examination (Feb 18 - ...  2022-02-18   
85   Mid-week Worship Service Examination (Jan 27 -...  2020-05-28   
191  Weekend Worship Service Examination (May 27 - ...  2022-05-27   
175  Weekend Worship Service Examination (Dec 24 - ...  2021-12-24   

     Total Comments                                           Post URL  
181             127  https://www.reddit.com/r/exIglesiaNiCristo/com...  
37               99  https://www.reddit.com/r/exIglesiaNiCristo/com...  
102              92  https://www.reddit.com/r/exIglesiaNiCristo/com...  
47               66  https://www.reddit.com/r/exIglesiaNiCristo/com...  
92               65  https://www.reddit.com/r/exIglesiaNiCristo/com...  
193              58  https://www.reddit.com/r/exIglesiaNiCristo/com...  
105              55  https://www.reddit.com/r/exIglesiaNiCristo/com...  
188              53  https://www.reddit.com/r/exIglesiaNiCristo/com...  
182              52  https://www.reddit.com/r/exIglesiaNiCristo/com...  
59               52  https://www.reddit.com/r/exIglesiaNiCristo/com...  
185              51  https://www.reddit.com/r/exIglesiaNiCristo/com...  
183              50  https://www.reddit.com/r/exIglesiaNiCristo/com...  
85               50  https://www.reddit.com/r/exIglesiaNiCristo/com...  
191              49  https://www.reddit.com/r/exIglesiaNiCristo/com...  
175              49  https://www.reddit.com/r/exIglesiaNiCristo/com...

## Top 5 WS Examination Posts on r/exIglesiaNiCristo:
1) https://www.reddit.com/r/exIglesiaNiCristo/comments/skfahc/weekend_worship_service_examination_feb_04_feb_07/
    
    - 127 comments
    
2) https://www.reddit.com/r/exIglesiaNiCristo/comments/shwyfu/midweek_worship_service_examination_feb_01_feb_03/

    - 99 comments
    
3) https://www.reddit.com/r/exIglesiaNiCristo/comments/v6wdhi/midweek_worship_service_examination_jun_07_jun_09/

    - 92 comments
    
4) https://www.reddit.com/r/exIglesiaNiCristo/comments/snm8t5/midweek_worship_service_examination_feb_08_feb_10/

    - 66 comments
    
5) https://www.reddit.com/r/exIglesiaNiCristo/comments/n05e43/midweek_worship_service_examination_apr_28_apr_30/

    - 65 comments